hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALFactory.java

```diff
@@ -311,7 +311,7 @@ public Reader createReader(final FileSystem fs, final Path path,
             reader.init(fs, path, conf, stream);
             return reader;
           }
-        } catch (IOException e) {
+        } catch (Exception e) {
```

Comment:
Catching `Exception` is enough? No need to catch `Throwable` to avoid any possibility of still getting this leak?